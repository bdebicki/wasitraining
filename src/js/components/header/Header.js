import { TITLE } from '../../enums/content';
import HEADER from '../../elementHandlers/header';
import LAYOUT from '../../elementHandlers/layout';
import { updateComponentByViewType } from '../../utils/updateView';
import Title from './Title';

export default class Header {
	constructor(data, target) {
		this.data = data;
		this.target = target;
	}

	static updateHeaderLocation(el) {
		updateComponentByViewType(el, HEADER.INTRO_HEADER_CLASS);
	}

	static renderHeaderContainer() {
		const headerContainer = document.createElement('header');

		headerContainer.id = LAYOUT.HEADER_ID;
		headerContainer.dataset.bgText = TITLE;
		Header.updateHeaderLocation(headerContainer);

		return headerContainer;
	}

	render() {
		const headerContainer = Header.renderHeaderContainer();

		headerContainer.appendChild(Title.render());

		document.getElementById(this.target).appendChild(headerContainer);
	}
}
