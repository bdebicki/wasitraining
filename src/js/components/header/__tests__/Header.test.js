import cleanDOM from '../../../../../tests/utils/cleanDOM';
import mockView from '../../../../../tests/utils/mockView';
import HEADER from '../elementHandlers/header';
import VIEW_TYPES from '../../../enums/viewTypes';
import Header from '../Header';

afterAll(() => cleanDOM());

describe('tests header component', () => {
	it('check dose title component renders correct', () => {
		// when
		const title = Header.render();

		// then
		expect(title).toMatchSnapshot();
	});

	it('check does title on home view has modifier class name', () => {
		// having
		mockView(VIEW_TYPES.INTRO);

		// when
		const title = Header.render();

		// then
		expect(title.classList.contains(HEADER.INTRO_HEADER_CLASS)).toBeTruthy();
	});
});
