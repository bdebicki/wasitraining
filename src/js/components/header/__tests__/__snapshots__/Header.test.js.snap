// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests header component check dose title component renders correct 1`] = `
<header
	data-bg-text="was it raining at the open'er festival?"
	id="header"
>
	<h1
		class="pageName"
	>
		<a
			class="pageName__link"
			href="#home"
		>
			was it raining at the open'er festival?
		</a>
	</h1>
</header>
`;
